raft.go

```diff
@@ -879,17 +887,23 @@ func (r *Raft) leaderLoop() {
 			r.setState(Follower)
 
 		case <-r.leaderState.commitCh:
-			// Get the committed messages
-			committed := r.leaderState.inflight.Committed()
-			for e := committed.Front(); e != nil; e = e.Next() {
-				// Measure the commit time
+			// Process the newly committed entries
+			commitIndex := r.leaderState.commitment.getCommitIndex()
+			r.setCommitIndex(commitIndex)
```

Comment:
Ugh all these index helpers use the name `term` for the argument. We should probably fix those use.